src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -439,10 +441,48 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 case 1:
                     assert(!instrIsRMW);
-                    if (targetReg != falseReg)
+
+                    if ((targetReg != falseReg))
                     {
-                        GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, falseReg);
+                        // If targetReg is not the same as `falseReg` then need to move
+                        // the `falseReg` to `targetReg`.
+
+                        if (op1->IsTrueAllMask())
+                        {
+                            // We already skip importing ConditionalSelect if op1 == trueAll, however
+                            // if we still see it here, it is because we wrapped the predicated instruction
+                            // inside ConditionalSelect.
+                            // As such, no need to move the `falseReg` to `targetReg`
+                            // because the predicated instruction will eventually set it.
+                            assert(intrin.op3->IsVectorZero());
+                            assert(falseReg == REG_NA);
+                        }
+                        else if (falseReg == REG_NA)
```

Comment:
Should this rather be a check against `op3->isContained()`?

Reply:
are there any cases we will miss if we have `falseReg == REG_NA` instead of `isContained()`?

Reply:
i already have the assert about `isContained()` inside.

Reply:
I'm unsure. It's just more typical, in my experience, to check the containment flag instead and so that's the thing I expected to happen here.

Reply:
fixed